src/components/ButtonWithDropdownMenu.js

```diff
@@ -79,7 +79,7 @@ function ButtonWithDropdownMenu(props) {
     const [popoverAnchorPosition, setPopoverAnchorPosition] = useState(null);
     const {windowWidth, windowHeight} = useWindowDimensions();
     const caretButton = useRef(null);
-    const selectedItem = props.options[selectedItemIndex];
+    const selectedItem = props.options[selectedItemIndex] || _.first(props.options);
```

Comment:
Now that I'm looking at it with more attention, you're right that this line of code is not explicit about that edge case. I like the idea of using the report state to change the behavior of the payment options here. Given the PR is already tested and approved, and also we're planning to make the Approve button "sticky" (it was not included here because it was considered a premature optimization), we can revise this later. Meanwhile, I will merge this PR so we can test the collect approvers flow more thoroughly.